src/pi_mutex.c

```diff
@@ -85,10 +85,9 @@ int pi_mutex_unlock(pi_mutex_t *mutex)
 	bool ret;
 
 	pid = gettid();
-#if 0
-	XXX
 	if (pid != (mutex->futex & FUTEX_TID_MASK))
-#endif
+		return EPERM;
+
```

Comment:
I was concerned this could result in a false positive, but the pthread_mutex_unlock() man page is clear that attempting to unlock a mutex the thread did not lock or that is unlocked already will result in either an error or in undefined behavior - both are satisfied here.